src/_pytest/pytester.py

```diff
@@ -550,12 +565,15 @@ def restore(self) -> None:
         sys.path[:], sys.meta_path[:] = self.__saved
 
 
-class Testdir:
+class TestPath:
     """Temporary test directory with tools to test/run pytest itself.
 
-    This is based on the ``tmpdir`` fixture but provides a number of methods
-    which aid with testing pytest itself.  Unless :py:meth:`chdir` is used all
-    methods will use :py:attr:`tmpdir` as their current working directory.
+    It provides an empty folder (accessible through the ``tmp_path`` attribute) where pytest
+    can be executed in isolation, and contains facilities to write tests, configuration files,
+    and match against expected output.
+
+    It attempts to isolate the test run from external factors as much as possible, modifying
+    the current  working directory to ``tmp_path`` and environment variables during initialization.
 
     Attributes:
 
```

Comment:
This commit should remove the "For backward compatibility, the read-only property ``tmpdir`` returns" comment below (can't comment on it directly)